src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenOptimizedNullableOperators.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    public class CodeGenOptimizedNullableOperators : CSharpTestBase
+    {
+        [Fact]
+        public void TestComparisonNullableWithNonDefaultConstantValue()
+        {
+            var code = @"
+public class C
+{
+    public bool M(int? x)
+    {
+        return x == 42;
+    }
+}
+";
+            CompileAndVerify(code).VerifyDiagnostics().VerifyIL("C.M", @"
```

Comment:
>CompileAndVerify(code) [](http://example.com/codeflow?start=12&length=22)

I think we should verify runtime behavior as well. Execute the code and test result of M with:
- null input
- 42
- not 42